### PR TITLE
Fix GC.GetGCMemoryInfo.TotalAvailableMemoryBytes 

### PR DESCRIPTION
**Pull request by naricc**
The current status is that Fix GC.GetGCMemoryInfo.TotalAvailableMemoryBytes (if no limit is set) indicates the total physical memory in the machine, not the physical memory available (after subtracting used memory). This fixes it.

### CODE REVIEW

**Comment by akoeplinger** [comment]
I assume this fixes https://github.com/dotnet/runtime/issues/2280 ?

File: src/mono/mono/sgen/sgen-memory-governor.c [hunk 1]
```diff
@@ -501,10 +501,10 @@ sgen_memgov_init (size_t max_heap, size_t soft_limit, gboolean debug_allowance,
 	sgen_register_fixed_internal_mem_type (INTERNAL_MEM_LOG_ENTRY, sizeof (SgenLogEntry));
 
 	if (max_heap == 0) {
-		sgen_gc_info.total_available_memory_bytes = mono_determine_physical_ram_size ();
+		sgen_gc_info.total_available_memory_bytes = mono_determine_physical_ram_available_size ();
```
Comment (BrzVlad) [inline src/mono/mono/sgen/sgen-memory-governor.c]: the available ram is ever changing

**Review by BrzVlad — CHANGES_REQUESTED** [review]
If we are at it, we should also fix the memory load stats which are incorrect.

I'm not sure what the high memory threshold should be, I suggest leaving it 0 for now, rather than have random confusing value.

**Comment by naricc** [comment]
> I assume this fixes #2280 ?

@akoeplinger   Actually its unrelated. I am still trying to repro that one.

File: src/mono/mono/utils/memfuncs.c [hunk 2]
```diff
@@ -292,4 +292,75 @@ mono_determine_physical_ram_size (void)
...
+	do {
+		ret = host_statistics(host, HOST_VM_INFO, (host_info_t)&vmstat, &count);
+	} while (ret == KERN_ABORTED);
+
+	if (ret != KERN_SUCCESS) {
+		g_warning ("Mono was unable to retrieve memory usage!");
+		return 0;
+	}
+
+	host_page_size(host, &page_size);
```
Comment (CoffeeFlux) [inline src/mono/mono/utils/memfuncs.c]:
```suggestion
	host_page_size (host, &page_size);
```

**Review by CoffeeFlux — COMMENTED** [review]
I find the `ifdef` spam here very difficult to read. If there are enough other functions with platform-specific behavior in here, I'd consider platform-specific source files (maybe Vlad has an opinion here?). Otherwise, I think it's worth at least commenting the endifs with the corresponding if to help make it easier to follow. We don't require that in the codebase as a whole, but for functions like this I think it adds a lot of value.

Also, I realize the code is copy-pasted, but I included some style nits anyway since they're easy to fix :)

**Comment by BrzVlad** [comment]
@CoffeeFlux I have no strong opinion on splitting this method up in other source files. I think it would be nice, but I wouldn't consider it as part of this PR.

**Comment by naricc** [comment]
> If we are at it, we should also fix the memory load stats which are incorrect.
> 
> I'm not sure what the high memory threshold should be, I suggest leaving it 0 for now, rather than have random confusing value.

Currently we are using the major_collection_trigger_size, and updating hte value every time we do a major GC. I didn't realize this was incorrect? I can make it zero.

File: src/mono/mono/sgen/sgen-memory-governor.c (hunk 1 above)
Comment (naricc) [reply src/mono/mono/sgen/sgen-memory-governor.c]: Ah, yeah, that was dumb of me.

**Comment by naricc** [comment]
I am abandoing this to do a PR that fixes all the stats at once.

**Pull request closed without merging**